src/flagpole/__init__.py

```diff
@@ -96,37 +97,60 @@ class Feature(BaseModel):
     name: constr(min_length=1, to_lower=True)  # type:ignore[valid-type]
     owner: constr(min_length=1)  # type:ignore[valid-type]
     segments: list[Segment]
+    """A list of segments to evaluate against the provided data"""
     enabled: bool = True
+    """Defines whether or not the feature is enabled."""
     created_at: datetime = Field(default_factory=datetime.now)
     """This datetime is when this instance was created. It can be used to decide when to re-read configuration data"""
+    context_builder: ContextBuilder | None = None
+    """Optional builder used to transform passed in data to an EvaluationContext whenever this feature is evaluated."""
+
+    def match(self, context_data: dict[str, Any]) -> bool:
+        if self.context_builder is not None:
+            context = self.context_builder.build(context_data)
+        else:
+            context = EvaluationContext(context_data)
```

Comment:
Could we have this be `match(self, context_data: EvaluationContext)`. The evaluation context can be generated by the feature handler, or something higher in the stack than the individual feature.